indexing-service/src/main/java/io/druid/indexing/overlord/TaskStorage.java

```diff
@@ -53,6 +53,13 @@
   public void addLock(String taskid, TaskLock taskLock);
 
   /**
+   * Upgrades the lock state in the storage facility to non-preemptive lock
```

Comment:
I think this should also throw an IllegalArgumentException if taskLock is preemptible -- callers should only be providing non-preemptible locks to this method